### PR TITLE
Федоров Матвей. Вариант 16. Технология STL. Сортировка Шелла с четно-нечетным слиянием Бэтчера.

### PR DESCRIPTION
**Pull request by Matthewsoul124**
# Сортировка Шелла с Четно-Нечетным Слиянием Бэтчера (STL)

## Описание
Алгоритм реализует двухэтапную сортировку:

1. **Сортировка Шелла**  
   Исходный массив сортируется с использованием параллельной версии алгоритма Шелла через `std::async`.

- **Вычисление промежутков (gaps)**  
  Используется последовательность Кнута:  
  `gap = (2^k - 1)`

- **Параллельная обработка**  
  Для каждого промежутка применяется параллельный цикл:
  ```cpp
  for (int group : groups) {
      futures.push_back(std::async(std::launch::async, [gap, n, &input_ref, group]() 
  });

2. **Слияние Бэтчера**  
   Отсортированный массив делится пополам, и части объединяются с использованием алгоритма Бэтчера.

### CODE REVIEW

File: tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/data/test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+100
```
Comment (petr1core) [inline tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/data/test.txt:1]: Remove this

File: tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+#include <memory>
+#include <string>
+#include <random>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+#include "core/util/include/util.hpp"
+#include "stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/include/ops_stl.hpp"
+
+TEST(fyodorov_m_shell_sort_with_even_odd_batcher_merge_stl, test_small_array) {
```
Comment (petr1core) [inline tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/func_tests/main.cpp:15]: Add two-element test, reverse negative array test and reverse negative duplicate-sequence test

**Review by petr1core — CHANGES_REQUESTED** [review]
And also add description, please

File: tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/func_tests/main.cpp [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+  std::vector<int> input = {35, 33, 42, 10, 14, 19, 27, 44};
```
Comment (KAPUSHA228) [inline tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/func_tests/main.cpp:16]: добавь тесты на обратную последовательность с дублями, например (3, 3, 2, 2, 1,1),
тест с частично отсортированным массивом,
тест с перемешанными чет и нечет,
тест с чередованием положительных и отрицательных,

**Review by KAPUSHA228 — CHANGES_REQUESTED** [review]

**Review by petr1core — APPROVED** [review]

File: tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/data/test.txt (hunk 1 above)
Comment (aobolensk) [reply tasks/stl/fyodorov_m_shell_sort_with_even_odd_batcher_merge/data/test.txt:1]: This comment is still not resolved

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Comment by Matthewsoul124** [comment]
@aobolensk all done

**Review by aobolensk — APPROVED** [review]
